coordinator/core/marbleapi.go

```diff
@@ -458,3 +474,42 @@ func (c *Core) setTTLSConfig(marble manifest.Marble, specialSecrets reservedSecr
 
 	return nil
 }
+
+// findUserSecrets finds all user-defined secrets required by a Marble
+func findUserSecrets(params *rpc.Parameters, secrets map[string]manifest.Secret) []string {
+	// Find the name of all secrets required by the marble
+	// For each match in a Argv/Env/File the sub-strings will be as follows:
+	// 0: the complete secret injection string ({{ <encoding> .Secrets.<name>.<part> }})
+	// 1: the encoding
+	// 2: the name of the secret
+	// 3: if specified the part of the secret, empty string otherwise
+	r := regexp.MustCompile("{{\\s*(raw|hex|base64|pem)\\s+\\.Secrets\\.([\\w-]+)(\\.[\\w-]+)?\\s*}}")
```

Comment:
Any chance to use something from the `template` package to do this job? @m1ghtym0 @Nirusu

Reply:
Regarding extracting the actions from a parsed template, there is this stackoverflow discussion: https://stackoverflow.com/questions/40584612/how-to-get-a-map-or-list-of-template-actions-from-a-parsed-template

If we were to implement it this way we would have to extract the actions from the parsed template for each Arg/Env/File, then check for only the actions requesting a value from `.Secrets`. The later step most likely being done with regex, at which point we might as well just pass over the entire string with regex and save us the template parsing

Reply:
Can we revert the changes in marbleapi.go and storewrapper.go and just don't accept empty values in EncodeSecretDataToRaw and EncodeSecretDataToPem?

Reply:
I made a pass over the template functions., they now take care of returning errors when requested secret is not set.
This removes most of the changes to `marbleapi.go`.
`getSecretMap` from `storewrapper.go` now returns a map of all non unique secrets, including user defined ones.
If a secret is not yet set, the function will return an empty secret for that key.